### PR TITLE
Sb 14036

### PR DESCRIPTION
**Pull request by amit-tarento**

### CODE REVIEW

File: actors/user/src/main/java/org/sunbird/user/actors/ResetPasswordActor.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.services.sso.SSOManager;
+import org.sunbird.services.sso.SSOServiceFactory;
+
+@ActorConfig(
+		  tasks = {"resetPassword"},
+		  asyncTasks = {}
+		)
+public class ResetPasswordActor extends BaseActor {
```
Comment (manzarul) [inline actors/user/src/main/java/org/sunbird/user/actors/ResetPasswordActor.java]: Java doc.

File: actors/user/src/main/java/org/sunbird/user/actors/ResetPasswordActor.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    
+	@Override
+	public void onReceive(Request request) throws Throwable {
+		String userId = (String)request.get(JsonKey.USER_ID);
+		String password = (String)request.get(JsonKey.PASSWORD);
+		resetPassword(userId,password);	
+	}
+	
+	private void resetPassword(String userId, String password) {
+		ProjectLogger.log("ResetPasswordActor:resetPassword: method called.",LoggerEnum.INFO.name());
```
Comment (manzarul) [inline actors/user/src/main/java/org/sunbird/user/actors/ResetPasswordActor.java]: Do userId and password basic validation before passing to keyclaok , if not valid then logs info and send 400 client error.

File: actors/user/src/main/java/org/sunbird/user/actors/ResetPasswordActor.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		if(ssoManager.updatePassword(userId, password)){
+			Response response = new Response();
+			response.put(JsonKey.RESPONSE, JsonKey.SUCCESS);
+		} else {
+			ProjectCommonException.throwServerErrorException(ResponseCode.invalidUserCredentials);
```
Comment (manzarul) [inline actors/user/src/main/java/org/sunbird/user/actors/ResetPasswordActor.java]: Add logger here. , Response should be server error not invalid credentials.

File: actors/user/src/main/java/org/sunbird/user/actors/ResetPasswordActor.java (hunk 2 above)
Comment (amit-tarento) [reply actors/user/src/main/java/org/sunbird/user/actors/ResetPasswordActor.java]: Already did in request validator

File: actors/user/src/main/java/org/sunbird/user/actors/ResetPasswordActor.java (hunk 3 above)
Comment (amit-tarento) [reply actors/user/src/main/java/org/sunbird/user/actors/ResetPasswordActor.java]: done

**Pull request merged**